pkg/ui/src/views/login/loginContainer.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { RouterState } from "react-router";
+
+import LoginPage from "src/views/login/loginPage";
+import Layout from "src/views/app/containers/layout";
+
+interface LoginContainerState {
+  loggedIn: boolean;
+}
+
+class LoginContainer extends React.Component<RouterState, LoginContainerState> {
+  constructor(props: RouterState) {
+    super(props);
+    // TODO(vilterp): the user might already have a valid session. how do we check that?
+    // maybe do some templating on the backend and add it in.
+    this.state = {
+      loggedIn: false,
+    };
+  }
+
+  handleLogin = () => {
+    this.setState({
+      loggedIn: true,
+    });
+  }
+
+  render() {
+    if (this.state.loggedIn) {
+      return (
+        <Layout {...this.props}>
```

Comment:
The general idea here is sound to me; with this method, we also don't need to remember any query params or URL paths, since this just renders in place regardless of the path.  I think it's a good place to start.

Reply:
@couchand said IRL he thinks this should redirect to `/login`, not just display the login page on whatever URL you're trying to go to. I think the argument is that with the redirect, we see `/login` in metrics.

Then we _do_ have to remember where you're trying to go somehow, with a GET parameter or something. (maybe it can be stuffed in the state of `LoginContainer` somehow, if that's still there).